daemon/volumes.go

```diff
@@ -121,7 +121,12 @@ func (daemon *Daemon) registerMountPoints(container *container.Container, hostCo
 				}
 				cp.Volume = v
 			}
-
+			if v, ok := mountPoints[m.Destination]; ok {
```

Comment:
These seven lines are the same three times; maybe we should have a helper for it.